File: Python/ByTier/A/2171_A_Shizuku_Hoshikawa_and_Farm_Legs.py
# Problem: CF 2171 A - Shizuku Hoshikawa and Farm Legs
# https://codeforces.com/contest/2171/problem/A

"""
Task: Count the number of valid configurations of chickens and cows given total number of legs.

Approach:
- Let x = number of chickens (2 legs each)
- Let y = number of cows (4 legs each)
- We have the equation: 2x + 4y = n
- Simplify: x + 2y = n/2
- For valid integer solutions, n must be even.
- If n is even, then x = n/2 - 2y
- y can range from 0 to n/4 (since x >= 0)
- So number of valid configurations = n/4 + 1

Time Complexity: O(1) - constant time per test case
Space Complexity: O(1) - only using a few variables
"""

for i in range(int(input())):  # Iterate over number of test cases
    n = int(input())  # Read number of legs
    if n & 1:  # Check if n is odd using bitwise AND
        print(0)  # If odd, no valid configuration exists
    else:
        print(n // 4 + 1)  # If even, calculate configurations


# https://github.com/VaHiX/CodeForces/

File: CPP/ByTier/A/2171_A_Shizuku_Hoshikawa_and_Farm_Legs.cpp
// Problem: CF 2171 A - Shizuku Hoshikawa and Farm Legs
// https://codeforces.com/contest/2171/problem/A

/*
 * Problem: Shizuku Hoshikawa and Farm Legs
 * 
 * Purpose: Given the total number of legs 'n', determine how many different combinations of chickens (2 legs) 
 *          and cows (4 legs) can produce exactly 'n' legs.
 * 
 * Algorithm/Techniques:
 *   - Mathematical analysis: 
 *     Let c = number of chickens, w = number of cows.
 *     Then, 2*c + 4*w = n => c = (n - 4*w)/2
 *     For valid integer solutions, (n - 4*w) must be even and non-negative.
 *     Since n is even (as shown by the condition), n-4*w is even if w is integer.
 *     Also, c >= 0 implies w <= n/4.
 *     So, valid values of w range from 0 to n/4 (inclusive).
 *     Thus, the count of valid configurations is floor(n/4) + 1.
 * 
 * Time Complexity: O(1) - constant time per test case
 * Space Complexity: O(1) - only using a fixed amount of extra space
 */

#include <iostream>

using namespace std;
int main() {
  int t;
  int n;
  cin >> t;
  for (int i = 0; i < t; i++) {
    cin >> n;
    if (n % 2 == 1) // If n is odd, no valid combination exists (since both chicken and cow have even legs)
      cout << "0\n";
    else
      cout << n / 4 + 1 << "\n"; // Number of valid combinations is floor(n/4) + 1
  }
  return 0;
}


// https://github.com/VaHiX/CodeForces/